### PR TITLE
[Attn] Add GPT-OSS-style attention sink support

### PR DESCRIPTION
**Pull request by Shomvel**
## Summary

- this PR adds GPT-OSS-style attention sink support to `parallel_attn`
- add a local `gpt-oss eager` sink reference that mirrors the `transformers` GPT-OSS sink path:
  - concatenate one sink logit per query head as an extra attention column
  - apply a single softmax over `[token_logits, sink_logit]`
  - drop the sink column before the value matmul
- verify that our sink reference in `naive_parallel_attn` is semantically aligned with the GPT-OSS eager implementation
- keep the runtime sink semantics consistent with the denominator-only sink update used in `sglang`

## Source

- `transformers` GPT-OSS implementation:
  - [`src/transformers/models/gpt_oss/modeling_gpt_oss.py](https://github.com/huggingface/transformers/blob/main/src/transformers/models/gpt_oss/modeling_gpt_oss.py)
- `sglang` reference:
  - GPT-OSS / MiMo sink plumbing in the model wrappers
  - denominator-only sink update in Triton decode / extend attention

## Float32 Error Summary

| case | output | dq | dk | dv | dsink |
| --- | ---: | ---: | ---: | ---: | ---: |
| `full` | `7.319450e-04` | `8.819103e-04` | `1.420081e-03` | `2.125859e-03` | `2.626896e-03` |
| `swa` | `6.992817e-04` | `7.513762e-04` | `1.618087e-03` | `2.214670e-03` | `2.434909e-03` |
| `indexed` | `6.468296e-04` | `6.054640e-04` | `7.749796e-04` | `4.768372e-07` | `2.619743e-03` |
| `indexed_swa` | `3.132820e-04` | `6.725043e-04` | `5.123019e-04` | `1.788139e-07` | `4.494488e-04` |
| `empty_row` | `0` | `0` | `0` | `0` | `0` |

## Summary by CodeRabbit

* **New Features**
  * Optional per-head "sink" logits for attention and decoding, plus optional gated-score input for gating behavior.

* **Improvements**
  * Revised gating behavior and online-softmax stabilization to handle empty-row/edge cases and keep merges finite.
  * Better shape validation and docstring corrections for gated inputs.

* **Tests**
  * Extensive tests validating sink-enabled attention across forward/backward paths, variable-length and sliding-window scenarios, and decoding.

### CODE REVIEW

File: fla/ops/attn/decoding.py [hunk 1]
```diff
@@ -100,6 +108,9 @@ def naive_attn_decoding_kernel(
         # [BT, BV]
         b_o = b_o * b_r + tl.sum(b_p[:, None] * b_v, 0)
         b_mp = b_m
+
+    if USE_SINK:
+        b_acc += exp(b_sink - b_m)
```
[the remark this answers is not in the document]
Comment (Shomvel) [reply fla/ops/attn/decoding.py]: I fixed this by matching the stabilization already used in `parallel.py`, and I also added a decoding test that covers a mixed varlen batch with one zero-length sequence.

File: fla/ops/attn/parallel.py [hunk 2]
```diff
@@ -686,7 +707,13 @@ def parallel_attn_bwd(
     dv = reduce(dv, 'b t (h g) v -> b t h v', g=G, reduction='sum')
     if g_cumsum is not None:
         dg_cumsum.add_(dg_cumsum_k)
-    return dq, dk, dv, dg_cumsum
+
+    dsinks = None
+    if sinks is not None:
+        p_sink = torch.exp2(sinks[None, None, :] - lse)
+        dsinks = -(p_sink * delta).sum((0, 1))
```
[the remark this answers is not in the document]
Comment (Shomvel) [reply fla/ops/attn/parallel.py]: I don't think this one needs an extra `RCP_LN2` multiply. The custom autograd function is already saving and differentiating with respect to the internally scaled sink tensor. If we multiply by `RCP_LN2` again inside the custom backward, we end up applying that scale twice.
